### PR TITLE
Setter for TextArea load_text

### PR DESCRIPTION
**Pull request by davetapley**
As discussed https://github.com/Textualize/textual/discussions/3525#discussioncomment-7277111

**Please review the following checklist.**

- [x] Docstrings on all new or modified functions / classes 
- [ ] Updated documentation
- [x] Updated CHANGELOG.md (where appropriate)

### CODE REVIEW

**Comment by davetapley** [comment]
Thoughts on tests / doc changes? 🤔

File: src/textual/widgets/_text_area.py [hunk 1]
```diff
@@ -929,6 +929,16 @@ def text(self) -> str:
         """The entire text content of the document."""
         return self.document.text
 
+    @text.setter
+    def text(self, value):
```
Comment (TomJGooding) [inline src/textual/widgets/_text_area.py:933]:
```suggestion
    def text(self, value: str) -> None:
```

File: src/textual/widgets/_text_area.py [hunk 2]
```diff
@@ -929,6 +929,16 @@ def text(self) -> str:
...
+        """Load text into the TextArea.
+
+        This will replace the text currently in the TextArea.
+
+        This is an alias of `load_text`.
```
Comment (TomJGooding) [inline src/textual/widgets/_text_area.py:938]:
```suggestion
        """Replace the text currently in the TextArea. This is an alias of `load_text`.
        
        Args:
            value: The text to load into the TextArea.
```

**Review by TomJGooding — COMMENTED** [review]
I'm not a maintainer of Textual, but your code is missing type annotations and I've suggested tweaks to the docstring.

Do you think you could also add a simple test?

**Comment by davetapley** [comment]
Thanks @TomJGooding, I wasn't sure about the argument doc string for a setter, but I guess there's no harm.

And r.e. tests it seems needlessly duplicitous when we already have `load_text` covered.
I suppose I could test that the setter calls that with a mock, but that's not really testing anything 🤪

**Comment by TomJGooding** [comment]
Sorry you might be right with this only being a setter. I think I was just on auto-pilot (and my change upset black).

I will mind my own business and leave this for the maintainers!

**Comment by darrenburns** [comment]
A simple test is still valuable as a guard against future regressions - it's a simple setter for now but could be modified in the future, so we want to capture it's expected behaviour. Just something like this would suffice:

```python
new_text = "hello\nworld\n"
text_area.text = new_text
assert text_area.text = new_text
```

**Comment by darrenburns** [comment]
I wrote this test here for it, but I think your fork is locked as I can't push to this PR:

```python
async def test_text_setter():
    app = TextAreaApp()
    async with app.run_test():
        text_area = app.query_one(TextArea)
        new_text = "hello\nworld\n"
        text_area.text = new_text
        assert text_area.text == new_text
```

Also, if you could add a CHANGELOG.md entry too when you get a moment, that'd be great.

**Review by darrenburns — APPROVED** [review]
Looks good, thanks!

**Pull request merged**